slither/core/declarations/contract.py

```diff
@@ -165,6 +166,14 @@ def is_library(self) -> bool:
     def is_library(self, is_library: bool):
         self._is_library = is_library
 
+    @property
+    def is_abstract(self) -> bool:
```

Comment:
I think we should either:
- Document that this supports only contracts where `abstract` is available
- Or implement an abstract contract check, to detect abstract contract before `abstract` was available 

Otherwise users might think that this will return true for any abstract contract, while it will only work for some

Reply:
Would it be a better idea to rename this property to "is_fully_implemented"?

The detector that will consume this property does not necessarily care about the `abstract` keyword; it just needs to be run on the "fully implemented" contract.

Reply:
yes I like this name. It matches `function.is_implemented`

Not that for the check, we need to filter out the functions that are shadowed  (`function. is_shadowed`) as the contract object has access to all the functions

Reply:
for the check, I am just using the `is_fully_implemented` attribute from the solidity AST, so I don't think we have to check for shadowed functions: https://github.com/ethereum/solidity/blob/7dd6d404815651b2341ecae220709a88aaed4038/libsolidity/ast/ASTJsonExporter.cpp#L295-L296